plugins/pencilblue/controllers/actions/user/sign_up.js

```diff
@@ -94,59 +95,31 @@ module.exports = function (pb) {
                 });
             }
 
-            var collection      = 'user';
             var successRedirect = '/user/login';
             var successMsg      = self.ls.g('login.ACCOUNT_CREATED');
             if(contentSettings.require_verification) {
-                collection      = 'unverified_user';
                 successRedirect = '/user/verification_sent';
                 successMsg      = self.ls.g('users.VERIFICATION_SENT') + post.email;
-                post.verification_code = util.uniqueId();
+                post.verificationCode = util.uniqueId();
             }
 
-            var user = pb.DocumentCreator.create(collection, post);
-
-            self.validateUniques(user, function(err, results) {
+            self.service.add(post, function(err, data) {
```

Comment:
We can't use the user service here.  The reason is because of a poor design decision we made early on with PB.  Unverified users are not in the same mongo collection as the rest of the users.  The user service only deals with verified (activated) users.  This is why the collection variable was used in the previous implementation.  It determined which collection to put the object in at runtime based on the content settings.  

We should revert this change.  The next version of PB (1.0) will make this breaking change so that we can simplify the logic as you have done by using the user service.